spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java

```diff
@@ -234,21 +243,39 @@ public void onChangeEvent(ConfigurationChangeEvent event) {
     /**
      * auto upgrade service detection
      */
-    private static void startDegradeCheck() {
-        executor.scheduleAtFixedRate(() -> {
-            if (degradeCheck) {
+    private void startDegradeCheck() {
+        executor.execute(() -> {
+            while (true) {
+                synchronized (this) {
+                    if (degradeCheck && !disable) {
+                        try {
+                            String xid = TransactionManagerHolder.get().begin(null, null, "degradeCheck", 60000);
+                            TransactionManagerHolder.get().commit(xid);
+                            onDegradeCheck(true);
+                        } catch (Exception e) {
+                            onDegradeCheck(false);
+                        }
+                    } else {
+                        try {
+                            this.wait();
+                        } catch (InterruptedException e) {
+                            if (LOGGER.isErrorEnabled()) {
+                                LOGGER.error("an unknown error occurred and the self - checking thread could not wait");
+                            }
+                            break;
+                        }
+                    }
+                }
                 try {
-                    String xid = TransactionManagerHolder.get().begin(null, null, "degradeCheck", 60000);
-                    TransactionManagerHolder.get().commit(xid);
-                    onDegradeCheck(true);
-                } catch (Exception e) {
-                    onDegradeCheck(false);
+                    Thread.sleep(degradeCheckPeriod);
+                } catch (InterruptedException e) {
+                    break;
                 }
             }
-        }, degradeCheckPeriod, degradeCheckPeriod, TimeUnit.MILLISECONDS);
+        });
     }
 
-    private static synchronized void onDegradeCheck(boolean succeed) {
+    private synchronized void onDegradeCheck(boolean succeed) {
```

Comment:
Locks are acquired for each transaction request, and I recommend reducing the scale or asynchrony of the locks

Reply:
> Locks are acquired for each transaction request, and I recommend reducing the scale or asynchrony of the locks

ok